nansat/vrt.py

```diff
@@ -918,22 +918,28 @@ def prepare_export_netcdf(self, options, bottomup):
         return options, add_gcps
 
     def copy(self):
-        """Create and return a full copy of a VRT instance"""
+        """Create and return a full copy of a VRT instance with new filenames"""
         if self.dataset.RasterCount == 0:
             vrt = VRT.from_gdal_dataset(self.dataset, metadata=self.dataset.GetMetadata())
         else:
             vrt = VRT.copy_dataset(self.dataset, metadata=self.dataset.GetMetadata())
-
-        vrt.band_vrts = dict(self.band_vrts)
+            replace_filenames = [(self.filename, vrt.filename)]
+            # recursive copy of vrt.vrt
+            if self.vrt is not None:
```

Comment:
Could that concept be easier? vrt from vrt and they all have apparently different filenames and all that. Is that really possible to trace that back in case of a crash?

Reply:
It is possible to trace it back. Every VRT has its filename both in the attribute `filename` in metadata, and each band refers to a specific filename. 
Easier? Probably.

Reply:
`vrt.vrt = self.vrt.copy()` this one is a very representative line. It is too complicated to understand what is going on here. Could you add more comments to the code?

Reply:
Will do.